chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/pendingListing/reviewer/edition2015/RequiredData2015Reviewer.java

```diff
@@ -323,10 +323,18 @@ public void review(final PendingCertifiedProductDTO listing) {
                                     try {
                                         Integer.valueOf(task.getTaskPathDeviationObserved());
                                     } catch (final Exception e) {
-                                        listing.getErrorMessages().add(
-                                                msgUtil.getMessage("listing.criteria.badTestTaskNumber",
-                                                        task.getUniqueId(), "Task Path Deviation Observed",
-                                                        task.getTaskPathDeviationObserved()));
+                                        try {
```

Comment:
Something about nested try/catch that I don't love.. I'm not sure if there is a java best-practice type reason against it other than maybe readability though. Would you consider changing this to have the outer try/catch where you try to parse the value as an Integer, then following the catch a check if the task.getTaskPathDeviationObserved value is null? If it's still null you could assume that the value didn't parse correctly as an integer and inside that if have the rest of the code that tries to parse as float, round, give warning.

Reply:
So the logic flow you're talking about is sort of
```
try parseInt // if successful, done
catch exception
  if val is null
    give "missing value error"
  else
    try round(parseFloat)
    give "rounding warning"
    catch exception
      give "could not parse error"
```


Our usual check workflow is sort of

```
if null
  give "missing value error"
else
  check logic of data
```

What I'm doing is expanding on that slightly

```
if null
  give "missing value error"
else
  try parseInt // if successful, done
  catch exception
    try round(parseFloat(
    give "rounding warning"
    catch exception
      give "could not parse error"
```

I think that's the same structure, but it moves the null check after the first parseInt thing. Given that we usually do null checks first I'd lean away from that strategy.

If I'm misinterpreting your logic flow please let me know

Reply:
I definitely should have used pseudocode.
`if null
  give "missing value error"
else
  try parseInt // if successful, done
  catch exception
      log error could not parse value as int

  if value is still null //it was not able to be parsed as an integer
      try round(parseFloat(
          give "rounding warning"
      catch exception
          give "could not parse error"`

Reply:
Ah, I see. The only issue with that option is that I'm not actually saving the value that is returned by the parseInt. I _could_, but it's not used anywhere else, so I didn't see a reason to. Without saving it I'm basically just jumping directly into the round(parseFloat step